src/iexamine.cpp

```diff
@@ -571,8 +565,35 @@ class atm_menu
                 return false;
             }
 
-            dst->charges += amount;
-            u.cash -= amount;
+            int inserted = 0;
+            int remaining = amount;
+            ammotype money( "money" );
+
+            std::sort( cash_cards_on_hand.begin(), cash_cards_on_hand.end(), []( item * one, item * two ) {
+                int balance_one = one->ammo_remaining();
+                int balance_two = two->ammo_remaining();
+                return balance_one > balance_two;
+            } );
+
+
+            for( auto const &cc : cash_cards_on_hand ) {
```

Comment:
```suggestion
            for( item * const &cc : cash_cards_on_hand ) {
```
Or some other way to replace the auto
```suggestion
            for( item *cc : cash_cards_on_hand ) {
```